src/backend/marsha/core/tests/test_api_livesession.py

```diff
@@ -4019,3 +3969,438 @@ def test_api_livesession_put_username_lti_already_exists(
+            display_name="John",
+            email="john@fun-test.fr",
+            is_registered=False,
+            lti_user_id="44444",
+            lti_id="Maths",
+            username="John",
+            video=video,
+        )
+
+        anonymous_live_session = LiveSessionFactory(
+            anonymous_id=uuid.uuid4(),
+            email=None,
+            is_registered=False,
+            video=video,
+        )
+
+        not_accessing_live_session = LiveSessionFactory(
+            consumer_site=video.playlist.consumer_site,
+            display_name="Sarah",
+            email="sarah@fun-test.fr",
+            is_registered=False,
+            lti_user_id="55555",
+            lti_id="Maths",
+            username="Sarah",
+            video=other_video,
+        )
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["consumer_site"] = str(video.playlist.consumer_site.id)
+        jwt_token.payload["context_id"] = "Maths"
+        jwt_token.payload["roles"] = [random.choice(["administrator", "instructor"])]
+        jwt_token.payload["user"] = {
+            "email": "admin@fun-test.fr",
+            "id": "11111",
+            "username": "Admin",
+        }
+
+        response = self.client.patch(
+            f"/api/livesessions/{live_session.id}/",
+            {"is_registered": True, "email": "r00t@fun-test.fr"},
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        live_session.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(live_session.email, "r00t@fun-test.fr")
+
+        response = self.client.patch(
+            f"/api/livesessions/{other_live_session.id}/",
+            {"is_registered": True, "email": "l33t@fun-test.fr"},
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        other_live_session.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(other_live_session.email, "l33t@fun-test.fr")
+
+        response = self.client.patch(
+            f"/api/livesessions/{anonymous_live_session.id}/",
+            {"is_registered": True, "email": "An0n@fun-test.fr"},
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        anonymous_live_session.refresh_from_db()
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(anonymous_live_session.email, "An0n@fun-test.fr")
+
+        response = self.client.patch(
+            f"/api/livesessions/{not_accessing_live_session.id}/",
+            {"is_registered": True, "email": "wr0ng@fun-test.fr"},
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+        )
+        self.assertEqual(response.status_code, 404)
```

Comment:
I think we should add a test with an update for an admin with an email already existing for a public registration

Reply:
done